### PR TITLE
Description attributes to resource structs

### PR DESCRIPTION
**Pull request by t0mk**
This PR copies attribute descriptions from docs to golang code, so that we can benefit from automatic doc generation.

Towards #13

### CODE REVIEW

**Comment by t0mk** [comment]
@displague this is the first step to generating docs

I would like this to be merged rather sooner than later, so that there's no drift.

When this is merged, we can start moving docs content to [conventional paths](https://github.com/hashicorp/terraform-plugin-docs#conventional-paths).

File: metal/datasource_metal_device_bgp_neighbors.go [hunk 1]
```diff
@@ -9,48 +9,58 @@ func bgpNeighborSchema() *schema.Resource {
...
+				Description: "Array of IP addresses of this neighbor's peers",
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
 			},
 			"routes_in": {
-				Type:     schema.TypeList,
-				Computed: true,
-				Elem:     bgpRouteSchema(),
+				Type:        schema.TypeList,
+				Description: "Array of incoming routes. Each route has attributes:",
```
Comment (displague) [inline metal/datasource_metal_device_bgp_neighbors.go:55]: was this a multiline description that got cut in the move?

File: metal/datasource_metal_device_bgp_neighbors.go [hunk 2]
```diff
@@ -60,12 +70,14 @@ func bgpRouteSchema() *schema.Resource {
...
-				Computed: true,
+				Type:        schema.TypeString,
+				Description: "CIDR expression of route (IP/mask)",
+				Computed:    true,
 			},
 			"exact": {
-				Type:     schema.TypeBool,
-				Computed: true,
+				Type:        schema.TypeBool,
+				Description: "(bool) Whether the route is exact",
```
Comment (displague) [inline metal/datasource_metal_device_bgp_neighbors.go:79]: "(bool) ", we won't need that.

File: metal/resource_metal_device_network_type.go [hunk 3]
```diff
@@ -20,12 +20,14 @@ func resourceMetalDeviceNetworkType() *schema.Resource {
...
-				Required: true,
-				ForceNew: true,
+				Type:        schema.TypeString,
+				Description: "The ID of the device on which the network type should be set",
+				Required:    true,
+				ForceNew:    true,
 			},
 			"type": {
 				Type:         schema.TypeString,
+				Description:  "Network type to set. Must be one of layer3, hybrid, layer2-individual and layer2-bonded",
```
Comment (displague) [inline metal/resource_metal_device_network_type.go:30]: Perhaps in the future, we can define lists like this as a const and then use a function to strings.Join (with "Must be one of ... or ...") for Description while reusing the const for ValidateFunc.

**Review by displague — APPROVED** [review]

**Comment by displague** [comment]
I merged #126 first and that introduced a conflict here.

After resolving that (and double-checking that any multiline descriptions are retained, like the one I left a comment about), this looks good to merge.

**Comment by t0mk** [comment]
@displague thanks for taking a look on this, it's a long diff and not very interesting.. I will resolve the conflict, fix your comment, check on more time and then merge.

File: metal/datasource_metal_device_bgp_neighbors.go (hunk 1 above)
Comment (t0mk) [reply metal/datasource_metal_device_bgp_neighbors.go:55]: yes

File: metal/resource_metal_device_network_type.go (hunk 3 above)
Comment (t0mk) [reply metal/resource_metal_device_network_type.go:30]: I can do it here and now.

**Pull request merged**